### PR TITLE
mes-7791 office alignment fix

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Fixed alignment issues on office page, moved eco card to its own component

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image
image
image

### CODE REVIEW

File: src/app/pages/office/cat-a-mod2/office.cat-a-mod2.module.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { FaultSummaryProvider } from '@providers/fault-summary/fault-summary';
 import {
   WaitingRoomToCarComponentsModule,
 } from '@pages/waiting-room-to-car/components/waiting-room-to-car.components.module';
+import { OfficeCatBPageModule } from '@pages/office/cat-b/office.cat-b.module';
```
Comment (matthew2564) [inline src/app/pages/office/cat-a-mod2/office.cat-a-mod2.module.ts]: Why would CatB need to be injected here?

The component should live inside ```office/components``` not the Cat B specific one

File: src/app/pages/office/components/eco-card/eco-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+@Component({
+  selector: 'eco-card',
+  templateUrl: './eco-card.component.html',
+  styleUrls: ['./eco-card.component.scss'],
+})
+export class ECOCardComponent {
+
+  @Input()
+  ecoFaults: any;
```
Comment (matthew2564) [inline src/app/pages/office/components/eco-card/eco-card.component.ts]: We should have types for both of these inputs

File: src/app/pages/office/cat-home-test/office.cat-home-test.page.html [hunk 3]
```diff
@@ -101,22 +101,10 @@ <h4 class="fault-heading">ETA</h4>
...
-                                </div>
-                            </ion-col>
-                        </ion-row>
-                    </ion-grid>
-                </ion-card-content>
-            </ion-card>
+            <eco-card
+                    [ecoDisplay]="pageState.displayEco$"
+                    [ecoFaults]="pageState.ecoFaults$"
+            ></eco-card>
```
Comment (matthew2564) [inline src/app/pages/office/cat-home-test/office.cat-home-test.page.html]: This should have an ```*ngIf``` around it like was there previously and then remove the if inside the component - the ```ecoDisplay``` input can then be completely removed

File: src/app/pages/office/components/eco-card/eco-card.component.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<ion-card id="eco" *ngIf="(ecoFaults | async) && (ecoDisplay | async)">
```
Comment (matthew2564) [inline src/app/pages/office/components/eco-card/eco-card.component.html]: This if can be removed if the if is housed at the higher level component

File: src/app/pages/office/components/eco-card/eco-card.component.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  <ion-card-content>
+    <ion-grid>
+      <ion-row class="ion-align-items-start mes-data">
+        <ion-col size="32">
+          <h4 class="fault-heading">ECO</h4>
+        </ion-col>
+        <ion-col class="padding-down ion-align-self-start ion-padding-start">
+          <div id="ecoFaults">{{ecoFaults | async}}</div>
```
Comment (matthew2564) [inline src/app/pages/office/components/eco-card/eco-card.component.html]: This should not need the async pipe, we should not be passing down observables into the component, we should be resolving them at the higher level

File: src/app/pages/office/cat-d/office.cat-d.module.ts [hunk 6]
```diff
@@ -35,6 +36,7 @@ import { OfficeCatDPageRoutingModule } from './office.cat-d-routing.module';
     PassFinalisationComponentsModule,
     TestFinalisationComponentsModule,
     WaitingRoomToCarComponentsModule,
+    OfficeCatBPageModule,
```
Comment (andrewsetterfield) [inline src/app/pages/office/cat-d/office.cat-d.module.ts]: rathern then adding your new component into officecatb module and injecting that everywhere, can you add it to the office.components module

**Review by matthew2564 — APPROVED** [review]

**Review by gittins7 — APPROVED** [review]

**Pull request merged**